src/redux/books/books.js

```diff
@@ -13,6 +16,65 @@ export const removeBook = (id) => ({
   id,
 });
 
+export const sentBookAPI = (payload) => (
+  async (dispatch) => {
+    await fetch(urlAPI, {
+      method: 'POST',
+      body: JSON.stringify({
+        item_id: payload.id,
+        title: {
+          bookTitle: payload.title,
+          bookAuthor: payload.author,
+        },
+        category: payload.category,
+      }),
+      headers: {
+        'Content-type':
+        'application/JSON',
+      },
+    });
+    dispatch(addBook(payload));
+  }
+);
+
+export const deleteBook = (id) => (
+  async (dispatch) => {
+    const response = await
+    fetch(`${urlAPI}/${id}`, {
+      method: 'DELETE',
+      body: JSON.stringify({
+        item_id: id,
+      }),
+      headers: {
+        'Content-type': 'application/JSON',
+      },
+    });
+    if (response.status === 201) {
+      dispatch(removeBook(id));
+    }
+  }
+);
+
+export const getBook = async (dispatch) => {
+  const response = await fetch(urlAPI);
+  const dataBooks = await
+  response.json();
+  const booksArr = Object.entries(dataBooks);
+  booksArr.forEach((bookItem) => {
+    const [id, value] = bookItem;
+    const { title, category } = value[0];
+    const { bookTitle, bookAuthor } = title;
+
+    const newBookEntry = {
+      id,
+      title: bookTitle,
+      author: bookAuthor,
+      category,
+    };
+    dispatch(addBook(newBookEntry));
+  });
+};
+
```

Comment:
- [ ] You should be making your API calls within redux actions not exporting them as regular callbacks then invoking them, kindly refactor your code so that data is persisted only with redux(using redux-thunk to return a function instead of an action).